### DISCUSSION TITLE
Newest gba repros

### DISCUSSION THREAD

**Discussion by alx5962** [sanni/cartreader#162] [Need help]
I have some gba repros bought few weeks ago I'd like to reflash. Each time I have the "Unknown Flash" error.

Here are the Flash ID codes displayed on the reader: EA3E, EA1F, EA1B, EA3D.

Some seem to use the Intel 4400 chipset, but I don't know how to confirm this. I took a picture of each card, as it may help identify them. Is there a database available somewhere identifying each one of them? 

![PXL_20210529_133525269](https://user-images.githubusercontent.com/1769807/120220441-02c21700-c23d-11eb-8853-91c3d81dcdce.jpg)
![PXL_20210529_133620849](https://user-images.githubusercontent.com/1769807/120220445-048bda80-c23d-11eb-8253-72b25039bd60.jpg)
![PXL_20210529_133744838](https://user-images.githubusercontent.com/1769807/120220452-06ee3480-c23d-11eb-8e80-3d9090b68fc8.jpg)
![PXL_20210529_133845239](https://user-images.githubusercontent.com/1769807/120220458-081f6180-c23d-11eb-9a9d-79074f7cebf0.jpg)
![PXL_20210529_134013120](https://user-images.githubusercontent.com/1769807/120220459-09508e80-c23d-11eb-8c4c-2e06aeb61901.jpg)

* In case some are using a 4400L0ZDQ0 variant. In the GBA.ino file, flashRepro_GBA() function, do I just replace "8816" with my own flash id codes to test if it works? Or Will I brick them If I do so ? 
 
 ```
 if ((strcmp(flashid, "8802") == 0) || (strcmp(flashid, "8816") == 0) || (strcmp(flashid, "227E") == 0)) {
 ...
 // Intel 4400L0ZDQ0
    else if (strcmp(flashid, "8816") == 0) {
      println_Msg(F("Intel 4400L0ZDQ0"));
    }
```
I'm new to this so I don't really know where to look at. Any help would be very appreciated.

**Comment by Androxilogin** [comment]
I also don't understand this process. It would be great if someone could make a YouTube video on how to do this or something.

**Comment by sanni** [comment]
Those are ST flashroms. The last picture is MX.  The MX29GL128 should work if you add the ID. The STs, I never had one of those so I can't comment. You have to read the datasheets and then write new flash functions accordingly. 

Or you can buy a GBxCart  RW, it focuses more on writing repros and already supports those chips: https://flashcartdb.com/index.php/Main_Page#Gameboy_Advance_Clone.2FRepo_Carts

A third option would be porting the code from the GBxCart  RW to the Cart Reader platform.
